L1Trigger/L1TTrackMatch/interface/DisplacedVertexProducer.h

```diff
@@ -0,0 +1,264 @@
+#ifndef __L1Trigger_L1TTrackMatch_DisplacedVertexProducer_h__
+#define __L1Trigger_L1TTrackMatch_DisplacedVertexProducer_h__
+
+#include "DataFormats/L1Trigger/interface/DisplacedVertex.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "PhysicsTools/ONNXRuntime/interface/ONNXRuntime.h"
+#include "SimTracker/TrackTriggerAssociation/interface/TTTrackAssociationMap.h"
+#include "TMath.h" 
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+#include <valarray>
+
+using namespace std;
+
+class Track_Parameters
```

Comment:
Is it necessary to create this new class? Can you just use the TTTrack class?

Reply:
It was convenient to define a class which contains the parameters of the projection of the track on the transverse plane just so I only calculate it once per track. Is there a preference for using functions or class methods for this type of thing?

Reply:
The preference is not necessarily for functions versus class members, but for condensing the number of in use classes to try and consolidate code and functionality. 

There are at least a few things conceptually similar between [`TTrack`](https://github.com/cms-sw/cmssw/blob/master/DataFormats/L1TrackTrigger/interface/TTTrack.h) and it's parent class [`TTTrack_TrackWord`](https://github.com/cms-sw/cmssw/blob/master/DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h) and this track parameters class. It would be good if the track trigger developers could try to centralize as many common elements to most types of tracks in one place if possible

Reply:
So what would you recommend? I could try inheriting from TTrack in my class and only add the extra functionality I need or I could try to do without an additional class at all.

Reply:
My recommendation would be to remove this class entirely, pass the L1 track into [these functions](https://github.com/cms-l1t-offline/cmssw/blob/0dde2bd9fc0af44d7cc964738620b957831502c0/L1Trigger/L1TTrackMatch/interface/DisplacedVertexProducer.h#L119-L142) and use the TTTrack/TTTrack_Word variables directly, and similarly for everywhere else this is used like [here](https://github.com/cms-l1t-offline/cmssw/blob/0dde2bd9fc0af44d7cc964738620b957831502c0/L1Trigger/L1TTrackMatch/interface/DisplacedVertexProducer.h#L212).